src/FlexView.tsx

```diff
@@ -81,81 +76,73 @@ export class FlexView extends React.Component<FlexView.Props> {
   }
 
   logWarnings(): void {
-    const {
-      basis,
-      shrink,
-      grow,
-      hAlignContent,
-      vAlignContent,
-      children,
-      column
-    } = this.props;
-
-    if (basis === "auto") {
-      warn(
-        'basis is "auto" by default: forcing it to "auto"  will leave "shrink:true" as default'
-      );
-    }
+    if (process.env.NODE_ENV !== "production") {
+      const {
+        basis,
+        shrink,
+        grow,
+        hAlignContent,
+        vAlignContent,
+        column
+      } = this.props;
+
+      if (basis === "auto") {
+        console.warn(
+          'basis is "auto" by default: forcing it to "auto"  will leave "shrink:true" as default'
+        );
+      }
 
-    if (
-      (shrink === false || shrink === 0) &&
-      (grow === true || (typeof grow === "number" && grow > 0))
-    ) {
-      warn('passing both "grow" and "shrink={false}" is a no-op!');
-    }
+      if (
+        (shrink === false || shrink === 0) &&
+        (grow === true || (typeof grow === "number" && grow > 0))
+      ) {
+        console.warn('passing both "grow" and "shrink={false}" is a no-op!');
+      }
 
-    if (
-      process.env.NODE_ENV !== "production" &&
-      typeof children !== "undefined" &&
-      !column &&
-      hAlignContent === "center"
-    ) {
-      const atLeastOneChildHasHMarginAuto = some(
-        [].concat(children as any),
-        (child: any) => {
-          const props =
-            (typeof child === "object" && child !== null
-              ? child.props
-              : undefined) || {};
-          const style = props.style || {};
-
-          const marginLeft = style.marginLeft || props.marginLeft;
-          const marginRight = style.marginRight || props.marginRight;
+      const children: Array<
+        | React.ReactElement<
+            { [k: string]: any } & { style?: React.CSSProperties }
+          >
+        | React.ReactText
+      > = React.Children.toArray(this.props.children);
+
+      if (!column && hAlignContent === "center") {
+        const atLeastOneChildHasHMarginAuto = some(children, child => {
```

Comment:
not related, and not sure about our current browser support target here, but this could easily be rewritten using `Array.prototype.some` and maybe drop some kB from the package?